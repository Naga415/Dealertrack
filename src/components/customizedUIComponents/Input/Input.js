import React from "react";
import PropTypes from "prop-types";
import "./Input.scss";

export default function Input({
  label,
  displayLabel,
  htmlId,
  addOnIndentation,
  addOnClassName,
  error,
  ...props
}) {
  return (
    <div className={`form-group ${error && 'has-error'}`}>
      {displayLabel && <label htmlFor={name}>{label}</label>}
      <div className="input-group">
        {!!addOnClassName &&
          addOnIndentation === "left" && (
            <span className="input-group-addon">
              <i className={addOnClassName} />
            </span>
          )}
        <input id={htmlId} name={name} type="text" className="form-control" {...props} />
        {error && <span className="form-control-feedback remove-background-color"/>}
      </div>
    </div>
  );
}

Input.propTypes = {
  addOnClassName: PropTypes.string,
  addOnIndentation: PropTypes.string,
  disabled: PropTypes.bool,
  displayLabel: PropTypes.bool,
  htmlId: PropTypes.string,
  label: PropTypes.string,
  name: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string.isRequired,
  error: PropTypes.bool
};

Input.default = {
  displayLabel: false
};
