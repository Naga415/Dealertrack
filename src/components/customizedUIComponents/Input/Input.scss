@import '~@coxautokc/fusion-theme/dist/fusion-variables.scss';

.form-group {
  span {
    cursor: not-allowed;
    background-color: #ebebea;
  }

  input[disabled] {
    cursor: not-allowed;
    background-color: #ebebea;
  }

  .remove-background-color {
    background: none;
  }
}
