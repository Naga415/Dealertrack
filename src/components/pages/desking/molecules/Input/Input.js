'use strict';
import React from 'react';
import PropTypes from 'prop-types';
import FormControl from 'react-bootstrap/lib/FormControl';
import DropdownButton from 'react-bootstrap/lib/DropdownButton';
import MenuItem from 'react-bootstrap/lib/MenuItem';
import ControlLabel from 'react-bootstrap/lib/ControlLabel';
import Button from 'react-bootstrap/lib/Button';

class Input extends React.Component {
  shouldComponentUpdate(nextProps){
    return this.props.value!== nextProps.value;
  }

  render(){
    let label = <ControlLabel className="desking-input__text-label">{this.props.label}</ControlLabel>;
    if (this.props.options) {
      label = (
        <DropdownButton
          id={this.props.id} bsSize="small"
          className="desking-input__dropdown"
          bsStyle="link"
          title={this.props.label}>
          {this.props.options.map((option, index)=><MenuItem key={index} eventKey={`${index}`}>{option.label}</MenuItem>)}
        </DropdownButton>);
    }
    else if (this.props.onClick) {
      label = <Button className="desking-input__link-label" bsStyle="link" onClick={this.props.onClick}>{this.props.label}</Button>;
    }

    return (
      <div className="desking-input">
        <div className="desking-input__label">
          {label}
        </div>
        <FormControl
          id = {this.props.id}
          name={this.props.name}
          className="desking-input__value"
          type="number"
          inputMode="numeric"
          placeholder={this.props.label}
          value={this.props.value || ''}
          onChange={this.props.onChange}
          onBlur={this.props.onBlur}
          bsSize="small"
          readOnly={this.props.readOnly}
          tabIndex={this.props.tabIndex}
        />
      </div>
    );
  }
}

Input.propTypes = {
  id:PropTypes.string.isRequired,
  name:PropTypes.string,
  label:PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  onBlur: PropTypes.func.isRequired,
  onClick: PropTypes.func,
  onChange: PropTypes.func,
  options: PropTypes.array,
  tabIndex: PropTypes.number,
  readOnly: PropTypes.bool
};

export default Input;
