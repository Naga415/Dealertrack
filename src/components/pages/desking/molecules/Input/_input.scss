$deskingInput:'.desking-input'; // namespace for component
$fontSize: 13px;

#{$deskingInput} {
  float: right;
  padding-bottom: 5px;

  &__text-label {
    vertical-align: sub;
    padding-right: 13px;
  }

  &__dropdown-label {
    font-size: $fontSize;
  }

  &__link-label {
    font-size: $fontSize;
  }

  &__label {
    float: left;
  }

  &__value {
    float: left;
    width: 100px;
    text-align: right;
  }
}
