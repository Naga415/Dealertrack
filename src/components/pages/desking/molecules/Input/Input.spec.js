/**
 * Created by RBANDERSON on 9/15/2017.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import Input from './Input';
import { mount} from 'enzyme';

describe('Tab', () => {
  const onChange = () => {

  };

  const blur = () => {

  };

  const onClickLink = () => {
 
  };

  const textLabelInputProps = {
    id:"TestId",
    name: "TestName",
    value: "TestValue",
    label: "TestLabel",
    onChange,
    onBlur: blur
  };

  const dropdownLabelInputProps ={...textLabelInputProps,...{options:[{label:1},{label:2}]}};

  const linkLabelInputProps ={...textLabelInputProps, ...{onClick:onClickLink}};

  function getInputNode(props) {
    return <Input {...props}/>;
  }
  
  function setupMount(props) {
    return mount(getInputNode(props));
  }

  it('renders <Input/>', () => {
    const component = renderer.create(getInputNode(textLabelInputProps));
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders label as text', () => {
    const component = setupMount(textLabelInputProps);
    expect(component.find('.desking-input__text-label').length).toBe(1);
  });

  it('renders label as dropdown', () => {
    const component = setupMount(dropdownLabelInputProps);
    expect(component.find('.desking-input__dropdown').length).toBe(1);
  });

  it('renders label as link', () => {
    const component = setupMount(linkLabelInputProps);
    expect(component.find('.desking-input__link-label').length).toBe(1);
  });
});
