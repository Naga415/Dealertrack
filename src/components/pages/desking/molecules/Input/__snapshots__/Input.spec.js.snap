// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tab renders <Input/> 1`] = `
<div
  className="desking-input"
>
  <div
    className="desking-input__label"
  >
    <label
      className="desking-input__text-label control-label"
      htmlFor={undefined}
    >
      TestLabel
    </label>
  </div>
  <input
    className="desking-input__value form-control input-sm"
    id="TestId"
    inputMode="numeric"
    name="TestName"
    onBlur={[Function]}
    onChange={[Function]}
    placeholder="TestLabel"
    readOnly={undefined}
    tabIndex={undefined}
    type="number"
    value="TestValue"
  />
</div>
`;
